src/platform/input/game-pads.js

```diff
@@ -74,56 +115,601 @@ const MAPS = {
             'PAD_R_STICK_X',
             'PAD_R_STICK_Y'
         ]
+    },
+
+    DEFAULT_XR: {
+        buttons: [
+            // Back buttons
+            'XRPAD_TRIGGER',
+            'XRPAD_SQUEEZE',
+
+            // Axes buttons
+            'XRPAD_TOUCHPAD_BUTTON',
+            'XRPAD_STICK_BUTTON',
+
+            // Face buttons
+            'XRPAD_A',
+            'XRPAD_B'
+        ],
+
+        axes: [
+            // Analogue Sticks
+            'XRPAD_TOUCHPAD_X',
+            'XRPAD_TOUCHPAD_Y',
+            'XRPAD_STICK_X',
+            'XRPAD_STICK_Y'
+        ]
     }
 };
 
 const PRODUCT_CODES = {
     'Product: 0268': 'PS3'
 };
 
+let deadZone = 0.25;
+const getGamepads = navigator ? navigator.getGamepads || navigator.webkitGetGamepads : undefined;
+
+/**
+ * A GamePadButton stores information about a button from the Gamepad API.
+ */
+class GamePadButton {
```

Comment:
I think we need some discussion on the naming here, the history so far is that we copy the exact name in `window` while using JSDoc like `/** @type {globalThis.GamepadButton} */` to distinguish. I am not quite happy with both ways, because I confuse these names easily and often wonder "what is it now?".

There are multiple ways for distinguishing:
1) GamepadButton vs GamePadButton (as you have it right now)
2) globalThis.GamepadButton vs GamepadButton (as in Mouse/Keyboard/Touch classes/events)
3) GamepadButton vs e.g. ExtendedGamepadButton (or any other prefix like "Higher...", "Advanced...")

Any thoughts on this @willeastcott since you handle most of JSDoc/typings?

Reply:
My point on having with `GamePadButton` is that the inital class was `GamePads` so I made everything consistent from the first naming

Reply:
Fine by me.